File: two_pointers/find_all_anagrams.rb
# From https://algo.monster/problems/find_all_anagrams

def find_all_anagrams(original, check)
    match_indexes = []
    
    left = 0
    right = check.size - 1
    while right <= original.size
        if original[left..right].chars.sort == check.chars.sort
            match_indexes << left
        end
        left += 1
        right += 1
    end

    match_indexes
end

File: two_pointers/find_all_anagrams.py
# From https://algo.monster/problems/find_all_anagrams

def find_all_anagrams(original: str, check: str) -> list[int]:
    left = 0
    right = len(check) - 1
    match_indexes = []

    while right <= len(original) - 1:
        if sorted(original[left:right + 1]) == sorted(check):
            match_indexes.append(left)
        left += 1
        right += 1
        
    return match_indexes